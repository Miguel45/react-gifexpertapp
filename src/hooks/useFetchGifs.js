import React from 'react'
import { getGift } from '../helpers/getGifs'

export const useFetchGifs = (category) => {

    const [state, setstate] = React.useState({
        data: [],
        loading: true
    });

    React.useEffect(() => {
        getGift(category).then(gifts => setstate({
            data: gifts,
            loading: false
        }))
    }, [category])
    
    return state
}