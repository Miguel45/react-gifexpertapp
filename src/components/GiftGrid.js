import React from 'react'
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'

import { GridItem } from './GridItem'

export const GiftGrid = ({ category }) => {

    const { data:images, loading } = useFetchGifs(category);

    return (
        <>
            <h3> {category} </h3>

            <h4 className="animate__animated animate__flash">{ loading && 'Cargando...' }</h4>
        
            <div className='card-grid'>
                {
                    images.map((img) => <GridItem key={img.id} {...img} />)
                }
            </div>
        
        </>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
}
