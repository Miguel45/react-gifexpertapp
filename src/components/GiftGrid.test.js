import React from 'react'; 
import { GiftGrid } from './GiftGrid';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { useFetchGifs } from '../hooks/useFetchGifs';

jest.mock('../hooks/useFetchGifs');

describe( 'Pruebas en el archivo GiftGrid.js', () => {

    const category = 'spiderman';

    //Enzyme
    test( 'Debe mostrarse bien el componente en un snapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow( <GiftGrid category={category}/> );

        expect( wrapper ).toMatchSnapshot();

    })

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        
        const gifs = [{
            id:'abc',
            url:'https://localhost/cualquiercosa/cosa.jpg',
            title: 'Cualquier cosa '
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow( <GiftGrid category={category}/> );
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GridItem').length).toBe(gifs.length)

    })

})