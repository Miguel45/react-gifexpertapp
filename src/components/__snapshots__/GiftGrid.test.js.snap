// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo GiftGrid.js Debe mostrarse bien el componente en un snapshot 1`] = `
<Fragment>
  <h3>
     
    spiderman
     
  </h3>
  <h4
    className="animate__animated animate__flash"
  >
    Cargando...
  </h4>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el archivo GiftGrid.js debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3>
     
    spiderman
     
  </h3>
  <h4
    className="animate__animated animate__flash"
  />
  <div
    className="card-grid"
  >
    <GridItem
      id="abc"
      key="abc"
      title="Cualquier cosa "
      url="https://localhost/cualquiercosa/cosa.jpg"
    />
  </div>
</Fragment>
`;
